File: test/asdf.c
#include <stdio.h>

#define BUFF_SIZE 10

void fill_buffer(char *buff){ 
    int i;
    for (i = 0; i < BUFF_SIZE; i++) {
        buff[i] = (char)i;
    }
}


int main(void) {
    char buff[BUFF_SIZE];

    fill_buffer(buff);

    int i;
    for (i = 0; i < BUFF_SIZE; i++) {
        printf("%d\n",buff[i]);
        // should print out 0 through 9
    }
    return 0;
}

File: test/asdf.rs
const BUFF_SIZE: usize = 10;

fn fill_buffer(buff: &mut [u8]){

    for i in 0..BUFF_SIZE {
        buff[i] = i as u8;
    }

}


fn main() {

    let mut buff: [u8;BUFF_SIZE] = [0u8;BUFF_SIZE];

    fill_buffer(&mut buff);

    println!("{:?}",buff);

}
